MiniApp/Classes/core/RealMiniApp.swift

```diff
@@ -212,6 +227,54 @@ internal class RealMiniApp {
     func cleanUpKeychain() {
         self.miniAppStatus.removeUnusedCustomPermissions()
     }
+
+    /// Method to check if all the required permissions mentioned in the manifest.json is agreed by the user.
+    /// - Parameters:
+    ///   - appId: MiniApp ID
+    ///   - versionId: Specific VersionID of a MiniApp
+    ///   - completionHandler: Handler that returns whether user agreed to required permissions or not.
+    func isRequiredPermissionsAllowed(appId: String, versionId: String, completionHandler: @escaping (Result<Bool, MASDKError>) -> Void) {
+        retrieveMiniAppMetaData(appId: appId, version: versionId) { (result) in
```

Comment:
The Host App should have already retrieved the manifest before calling `create`, so this means we will end up making two API requests for the exact same thing?

Maybe we could make a class for in-memory caching the manifest from the API? For example, when the Host App calls `getMiniAppManifest`, we will cache the manifest in-memory (not to a file). And then when they call `create`, we can check the in-memory cache whether we already downloaded the manifest for that version. If not, then we can call the API to download it. The Host App should typically only be calling `getMiniAppManifest` just before they call `create`, so we would probably only need to cache the latest manifest response from `getMiniAppManifest`.

What do you think @leorjoseph @khairul-alam-licon ?

Reply:
Yeah, in Android side, I was caching the latest manifest which is fetched by `getMiniAppManifest` from api. 
@leorjoseph Do you think, it's sound good?

Reply:
It's done already @khairul-alam-licon. Also, @leorjoseph is not my official ID. Please use @ts-leojoseph-r

Reply:
come on @leorjoseph !